src/Adapter/MSTestAdapter.PlatformServices/Extensions/ExceptionExtensions.cs

```diff
@@ -13,22 +13,18 @@ internal static class ExceptionExtensions
     /// </summary>
     /// <param name="exception"> The exception. </param>
     /// <returns> Custom exception message that includes inner exceptions. </returns>
-    internal static string GetExceptionMessage(this Exception? exception)
+    internal static string GetExceptionMessage(this Exception exception)
     {
-        if (exception == null)
-        {
-            return string.Empty;
-        }
-
-        string exceptionString = exception.Message;
+        var builder = new StringBuilder(exception.Message);
```

Comment:
I'm not sure if it's worth switching to `StringBuilder`.

- This code path is (hopefully) not hot and won't be performance sensitive. So, "hopefully" it won't matter that much whether or not we are using a `StringBuilder`.
- I'm thinking that in many cases, `InnerException` may actually be null. In this case we created a `StringBuilder` for nothing.

@Evangelink What do you think?

Reply:
i was also on the fence on this.

Reply:
I don't think there is much more bad in one or the other. I do think it's true that most of the time you don't have inner exceptions so the stringbuilder isn't adding a lot. But again it should be neglictible.